lang/golang/golang/Makefile

```diff
@@ -52,7 +57,7 @@ HOST_GO_VALID_OS_ARCH:= \
   linux_ppc64 linux_ppc64le linux_mips linux_mipsle linux_mips64 linux_mips64le
 
 BOOTSTRAP_SOURCE_PROTO:=git
-BOOTSTRAP_SOURCE_URL:=https://go.googlesource.com/go
+BOOTSTRAP_SOURCE_URL:=https://github.com/golang/go.git
```

Comment:
This pulls quite a bit of unnecessary data, use codeload instead?

Reply:
I haven't been able to find any official documentation for codeload.github.com, would prefer to not hook too deeply into GitHub's internals.

Also, my SDK test compiles have been downloading the bootstrap code from [sources.lede-project.org](https://sources.lede-project.org/golang-bootstrap-20170926-4d5426a5.tar.xz). I believe once the buildbots have cloned the repo once, they create and upload the xz archive so they (and we) don't have to do a full checkout again.

Reply:
The downside is that every client will first try git before falling back to tarball afaik. There are a few packages in in the package tree using codeload just fine. Just do a grep for codeload.

Reply:
how about we update the git cloning code to be smarter then, instead of saying, "Here's how you clone a repo" "oh, but no, you're not actually allowed to use it though"  Because that's just being rude to users.

Reply:
@diizzyy I believe all clients [attempt to download from a mirror](https://github.com/openwrt/openwrt/blob/4700544e4068cb72932148ac1ecd294ca1388671/include/download.mk#L161) before falling back to a git checkout.

Reply:
The file it downloads from sources.lede-project.org is a tarball ([link](https://sources.lede-project.org/golang-bootstrap-20170926-4d5426a5.tar.xz)).